controllers/edgedevice_controller.go

```diff
@@ -61,6 +68,28 @@ func (r *EdgeDeviceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	logger.Info("Reconciling", "edgeDevice", edgeDevice)
 
+	// create object bucket claim for edge-device
+	if edgeDevice.Status.DataOBC == nil || len(*edgeDevice.Status.DataOBC) == 0 {
+		obc, err := r.Claimer.GetClaim(ctx, edgeDevice.Name, edgeDevice.Namespace)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				logger.Info("Failed to find an existing OBC for the device. Creating new OBC", "edgeDevice", edgeDevice)
+				obc, err = r.Claimer.CreateClaim(ctx, edgeDevice)
+				if err != nil {
+					logger.Error(err, "Cannot create object bucket claim for the device", "EdgeDevice Name", edgeDevice.Name, "EdgeDevice Namespace", edgeDevice.Namespace)
+					return ctrl.Result{Requeue: true}, err
+				}
+			} else {
+				logger.Error(err, "Failed to get OBC for the device", "edgeDevice", edgeDevice)
```

Comment:
Shouldn't `ctrl.Result{Requeue: true}, err` be returned in this case? `obc` would be `nil` and as a result line 87 would panic.

Reply:
Fixed.